lib/weka/core/instances.rb

```diff
@@ -266,6 +307,50 @@ def map_attribute_type(type)
         return -1 unless Attribute::TYPES.include?(type.downcase.to_sym)
         Attribute.const_get(type.upcase)
       end
+
+      # Convert a hash whose keys are attribute names and values are attribute
+      #   values into an array containing attribute values in the order
+      #   of the Instances attributes.
+      #
+      # @param [Hash] attribute_values a hash whose keys are attribute
+      #   names and values are attribute values.
+      #
+      # @return [Array] an array containing attribute values in the
+      #   correct order
+      def attribute_values_from_hash(attribute_values)
+        names = attribute_names(true).map!(&:to_sym)
+        names.inject([]) do |values, attribute_name|
+          values << attribute_values[attribute_name]
+        end
```

Comment:
Instead of the names.inject block you can use `Hash#fetch_values`:
```ruby
attribute_values.fetch_values(*names)
```
Also, should attribute_values be named hash? For me it sounds more like an array just from its name...